content/2022/prepoznavanje-govora.md

```diff
@@ -2,3 +2,250 @@
+
+Svako stablo odlučivanja će dati svoj rezultat, a onaj rezultat koji se najviše puta pojavi biće izabran kao konačno predviđanje celog klasifikatora.
+
+![Random Forest](static\images\RandomForest1.svg)
+
+Pošto su pojedinačna stabla veoma osetljiva na podatke koji im se pruže, koristi se **Bagging** (ili **B**ootstrap **Agg**regat**ing**) princip. On dozvoljava dve bitne stvari:
+
+1.	Svakom stablu da nasumično izabere podatke sa kojima će da radi iz baze i time znatno smanji mogućnost overfitting-a.
+
+2.	Svako stablo dobija neki nasumičan feature na kom će se trenirati, umesto da se trenira na skupu feature-a, što bi zahtevalo i veću dubinu mreže. Ovaj aspekt, zvani Random Subspace Method ili Attribute Bagging, smanjuje korelaciju između stabala i time ih čini nezavisnijim jedne od drugih.
+
+##### 4. XGBoost
+
+XGBoost (Gradient Boosted Trees), kao i Random Forest, koristi više stabala odlučivanja za predviđanje i labeliranje. 
+
+Razlika između ova dva metoda može se primetiti u samom imenu: XGBoost koristi dodatnu metodu za predviđanje koja se zove Boosting. Boosting kombinuje slabija drva kako bi, ispravljajući njihove greške, sačinio nova drva sa što boljim rezultatima. Početna drva nazivaju se panjevi, i oni se sastoje od jednostavnih DA/NE odgovora za predskazanje.
+
+Dodatak Boosting-u ogleda se u loss funkciji. Cost funkcija (kriterijumska funkcija) jeste usrednjena vrednost svih funkcija greške (loss function), a loss funkcija je funkcionalna veza željenog outputa i dobijenog outputa u funkciji.
+
+Najkorišćenija loss funkcija je Cross Entropy Loss. 
+Cross Entropy Loss radi tako što pokušava da minimizuje razliku između tačnih rezultata i verovatnoće predviđanja, to jest output.
+
+Formula po kojoj se računa Cross Entropy Loss je sledeća:
+
+$L_{C E}(\hat{y}, y)=-[y \log \sigma(\mathbf{w} \cdot \mathbf{x}+b)+(1-y) \log (1-\sigma(\mathbf{w} \cdot \mathbf{x}+b))]$
+
+XGBoost se u Pythonu implementira bibliotekom xgboost. 
+
+##### 5. SVM
+
+Posao SVM klasifikatora je da u N-dimenzionalnom prostoru, gde je N broj parametara, pronađe hiperravan koja na najbolji način klasifikuje sve tačke koje predstavljaju podaci.
+
+Kako postoji velik broj ovih hiperravni, kao optimalnu uzimamo onu kod koje je udaljenost granice odlučivanja podjednako udaljena od podataka svih tipova. Ovim dobijamo veću verovatnoću da će bilo koji naknadno dodati podatak biti pravilno klasifikovan. 
+
+Hiperravni koje ograničavaju zonu udaljenosti od granice odlučivanja na kojoj klasifikator daje vrednosti čija je apsolutna vrednost manja od 1 nazivaju se noseći vektori. To znači da za svaki podatak koji se nalazi unutar tih vektora ne možemo sa sigurnošću reći kojoj klasi pripada.
+
+![SVM1](static/images/SVM.svg)
```

Comment:
Neke stvari koje mi se ne sviđaju kod ovog grafika:
* Mnogo različitih debljina linija. Probajte da se držite jedne debljine, eventualno nešto podebljate/stanjite da bi dodali na značaju
* Veličine kružića i kvadratića su prevelike
* Font je mnogo mali
* Debljina linija na osama je preveliki
* Ove linije koje ste obeležili sa "vektori podrške" nisu vektori podrške. Prvo, označili ste dve linije kao "vektor", a linije nisu vektori. Drugo, "vektor podrške" (pretpostavljam da mislite na support vector) je normalan na hipotezu (liniju koja vrši spearaciju klasa). Linije možete nazvati "margine", ali proverite sa Natašom i Pađinom.
* Margine (isprekidane linije) treba da prolaze kroz sample-ove, a ne pored njih. Dakle, treba da prolaze kroz centre kružića i kvadratića

Evo dobrih primera SVM objašnjenja:
https://upload.wikimedia.org/wikipedia/commons/thumb/7/72/SVM_margin.png/1280px-SVM_margin.png
https://static.javatpoint.com/tutorial/machine-learning/images/support-vector-machine-algorithm.png
https://miro.medium.com/max/720/0*rIGWbxfCPvxr_cz1